tree/dataframe/src/RDatasetGroup.cxx

```diff
@@ -33,6 +33,10 @@ RDatasetGroup::RDatasetGroup(const std::string &groupName,
                              const RMetaData &metaData)
    : fGroupName(groupName), fMetaData(metaData)
 {
+   // avoid constructing and destructing the helper TChain here if we don't need to
+   if (treeAndFileNameGlobs.empty())
+      return;
+
```

Comment:
IIUC the only purpose of the helper TChain is to expand the names of files that can be passed via globs. This probably calls for a refactoring of the globbing in TChain, moving it out of the constructor and into separate free functions. This way, if the user passes an empty `treeAndFileNameGlobs` argument, no parsing would be done automatically

Reply:
sure, for another PR I'd say :)